samples/drivers/led_ws2812b_sw/src/main.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <string.h>
+
+#define SYS_LOG_LEVEL SYS_LOG_LEVEL_DEBUG
+#include <logging/sys_log.h>
+
+#include <zephyr.h>
+#include <led_strip.h>
+#include <device.h>
+#include <misc/util.h>
+
+/*
+ * Number of RGB LEDs in the LED strip, adjust as needed.
+ */
+#define STRIP_NUM_LEDS 24
+
+#define STRIP_DEV_NAME CONFIG_WS2812B_SW_NAME
+#define DELAY_TIME K_MSEC(40)
+
+static const struct led_rgb colors[] = {
+	{ .r = 0xff, .g = 0x00, .b = 0x00, }, /* red */
+	{ .r = 0x00, .g = 0xff, .b = 0x00, }, /* green */
+	{ .r = 0x00, .g = 0x00, .b = 0xff, }, /* blue */
+};
+
+static const struct led_rgb black = {
+	.r = 0x00,
+	.g = 0x00,
+	.b = 0x00,
+};
+
+struct led_rgb strip_colors[STRIP_NUM_LEDS];
+
+const struct led_rgb *color_at(size_t time, size_t i)
+{
+	size_t rgb_start = time % STRIP_NUM_LEDS;
+
+	if (rgb_start <= i && i < rgb_start + ARRAY_SIZE(colors)) {
+		return &colors[i - rgb_start];
+	} else {
+		return &black;
+	}
+}
+
+void main(void)
+{
+	struct device *strip;
+	size_t i, time;
+
+	strip = device_get_binding(STRIP_DEV_NAME);
```

Comment:
Can't the `led_ws2812` sample be used instead, just not building the call/check to `device_get_binding(SPI_DEV_NAME)`? Apart from that (and the number of LEDs), the samples are identical.

Reply:
Yes, I could add some #ifdefs to the existing sample instead of adding a new one.

Reply:
I've completed the changes locally, but I noticed that this breaks the pattern a little bit: it seems all LED drivers have essentially a copy of the same sample if you look at `samples/drivers/led_*`

Reply:
I.e. if we start adding `#ifdefs` maybe all drivers should be collected into the same sample (however in that case the name of the sample starts being misleading since it's not ws2812 anymore)

Reply:
I've pushed now an extension of the existing sample instead of adding a new one. Let me know how that looks. I also had to remove any SPI & ws2812 driver stuff from prj.conf and move them to the board/*.conf.